### ISSUE TITLE
Components organisation

### ISSUE THREAD

**Issue by vvo** [algolia/instantsearch.js#57]
Right now we have:

![2015-09-03-140553_262x457_scrot](https://cloud.githubusercontent.com/assets/123822/9657846/f8ff021e-5244-11e5-96d7-ff6768353717.png)

It always feels to me too much folders, don't you think we could flatten this like:

- BemHelper.js
- Hits.js
- MultipleChoiceList.js
- Pagination.js
- PaginationHiddenLink.js
- PaginationLink.js
- Paginator.js
- Searchbox.js
- Template.js

So that we remove lot of folders and do not have that much of index.js files.

If at some point a component needs external files, then we can make a folder for this case.

**Comment by pixelastic** [comment]
I agree having all the files named `index.js` made things a bit complex at first. If this is not a React requirement, I'll also be in favor of using files instead of directories

**Comment by bobylito** [comment]
Yes seems more practical :+1: (and makes fuzzy search easier :) )

**Comment by redox** [comment]
LGTM :+1:
